pkg/compactor/compactor_test.go

```diff
@@ -1192,3 +1193,88 @@ func TestAllowedUser(t *testing.T) {
 		})
 	}
 }
+
+func TestCompactor_DeleteLocalSyncFiles(t *testing.T) {
+	numUsers := 10
+
+	// Setup user IDs
+	userIDs := make([]string, 0, numUsers)
+	for i := 1; i <= numUsers; i++ {
+		userIDs = append(userIDs, fmt.Sprintf("user-%d", i))
+	}
+
+	inmem := objstore.NewInMemBucket()
+	for _, userID := range userIDs {
+		id, err := ulid.New(ulid.Now(), rand.Reader)
+		require.NoError(t, err)
+		require.NoError(t, inmem.Upload(context.Background(), userID+"/"+id.String()+"/meta.json", strings.NewReader(mockBlockMetaJSON(id.String()))))
+	}
+
+	// Create a shared KV Store
+	kvstore := consul.NewInMemoryClient(ring.GetCodec())
+
+	// Create two compactors
+	var compactors []*Compactor
+
+	for i := 1; i <= 2; i++ {
```

Comment:
The two compactors should run with a different `DataDir` and the dir each compactor should be a temporarily dir removed once the tests ends.

Reply:
`prepare` function used in the loop already does that.